src/go/rpk/pkg/cli/debug/bundle/bundle.go

```diff
@@ -182,6 +185,43 @@ func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 	return cmd
 }
 
+func isK8sEnv() bool {
+	// To execute the appropriate bundle we look for
+	// kubernetes_service_* env variables to identify if we are in a
+	// k8s environment.
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	return !(len(host) == 0 || len(port) == 0)
+}
+
+func isDockerEnv(fs afero.Fs) bool {
+	// https://stackoverflow.com/a/20012536
+	bs, err := afero.ReadFile(fs, "/proc/1/cgroup")
+	if err != nil {
+		return false
+	}
+	s := string(bs[:])
+	if strings.Contains(s, "docker") || strings.Contains(s, "lxm") {
+		return true
+	}
+	return false
+}
+
+func getRedpandaPID(fs afero.Fs) int {
+	if isK8sEnv() || isDockerEnv(fs) {
+		return 1
+	}
```

Comment:
Doesn't `pgrep redpanda` return `1` in these environments?

Reply:
It should/would, but we don't have it in our docker image :(